x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportPutFollowAction.java

```diff
@@ -131,98 +127,95 @@ private void createFollowerIndex(
             return;
         }
 
-        ActionListener<Boolean> handler = ActionListener.wrap(
-                result -> {
-                    if (result) {
-                        initiateFollowing(request, listener);
-                    } else {
-                        listener.onResponse(new PutFollowAction.Response(true, false, false));
-                    }
-                },
-                listener::onFailure);
-        // Can't use create index api here, because then index templates can alter the mappings / settings.
-        // And index templates could introduce settings / mappings that are incompatible with the leader index.
-        clusterService.submitStateUpdateTask("create_following_index", new AckedClusterStateUpdateTask<Boolean>(request, handler) {
+        String remoteCluster = request.getRemoteCluster();
 
-            @Override
-            protected Boolean newResponse(final boolean acknowledged) {
-                return acknowledged;
-            }
+        Client client = CcrLicenseChecker.wrapClient(this.client, threadPool.getThreadContext().getHeaders());
 
-            @Override
-            public ClusterState execute(final ClusterState currentState) throws Exception {
-                String followIndex = request.getFollowRequest().getFollowerIndex();
-                IndexMetaData currentIndex = currentState.metaData().index(followIndex);
-                if (currentIndex != null) {
-                    throw new ResourceAlreadyExistsException(currentIndex.getIndex());
-                }
+        final ActionListener<PutFollowAction.Response> followingListener;
+        if (request.getWaitForCompletion()) {
+            followingListener = listener;
+        } else {
+            followingListener = ActionListener.wrap(() -> {});
+            listener.onResponse(new PutFollowAction.Response(false, false, false));
```

Comment:
We should at least make sure that the restore is properly kicked off, i.e., we should at least make it to the place where you're calling `createAndRegisterListener`, because otherwise you're returning a follower response for an index that's not even been created.